language/move-vm/runtime/src/loader.rs

```diff
@@ -537,23 +539,40 @@ impl Loader {
         ty_args: &[TypeTag],
         data_store: &mut impl DataStore,
     ) -> VMResult<(Arc<Function>, Vec<Type>)> {
-        self.load_module_expect_no_missing_dependencies(module_id, data_store)?;
-        let idx = self
-            .module_cache
+
+        {
+            let _timer = LIBRA_MOVEVM_LOAD_MODULE_EXPECT_NO_MISSING_DEPENDENCIES_SECONDS.start_timer();
+            self.load_module_expect_no_missing_dependencies(module_id, data_store)?;
+        }
+
+        let idx = {
+            let _timer = LIBRA_MOVEVM_MODULE_CACHE_RESOLVE_SECONDS.start_timer();
+            self.module_cache
             .lock()
             .unwrap()
             .resolve_function_by_name(function_name, module_id)
-            .map_err(|err| err.finish(Location::Undefined))?;
-        let func = self.module_cache.lock().unwrap().function_at(idx);
+            .map_err(|err| err.finish(Location::Undefined))?
+        };
+
+        let func = {
+            let _timer = LIBRA_MOVEVM_MODULE_CACHE_SECONDS.start_timer();
+            self.module_cache.lock().unwrap().function_at(idx)
+        };
 
         // verify type arguments
         let mut type_params = vec![];
-        for ty in ty_args {
-            type_params.push(self.load_type(ty, data_store)?);
+        {
+            let _timer = LIBRA_MOVEVM_LOAD_TYPE_SECONDS.start_timer();
```

Comment:
this is another questionable timer. First time we load the types it will be an expensive operation, otherwise it should be minimal